paasta_tools/kubernetes_tools.py

```diff
@@ -1679,6 +1689,9 @@ def get_pod_template_spec(
             # character limit for k8s labels (63 chars)
             labels["paasta.yelp.com/deploy_group"] = self.get_deploy_group()
 
+        if piscina_autoscaling_configured:
```

Comment:
super-nit: imo (and feel free to drop if you disagree) switching to `elif`
```suggestion
        elif piscina_autoscaling_configured:
```
would be a little clearer here: only one of these conditionals should ever be true